### PR TITLE
Update Redis Docs

### PR DESCRIPTION
**Pull request by AbdullahAlShaad**
Signed-off-by: Shaad7 <abdullah.alshaad@appscode.com>

### CODE REVIEW

File: docs/guides/redis/concepts/redisopsrequest.md [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+    kubectl.kubernetes.io/last-applied-configuration: |
+      {"apiVersion":"ops.kubedb.com/v1alpha1","kind":"RedisOpsRequest","metadata":{"annotations":{},"name":"redisops-vertical","namespace":"demo"},"spec":{"databaseRef":{"name":"standalone-redis"},"type":"VerticalScaling","verticalScaling":{"redis":{"limits":{"cpu":"500m","memory":"800Mi"},"requests":{"cpu":"200m","memory":"300Mi"}}}}}
+  creationTimestamp: "2023-02-02T09:14:01Z"
+  generation: 1
+  name: redisops-vertical
+  namespace: demo
+  resourceVersion: "483411"
+  uid: 12c45d9c-daea-472d-be61-b88505cb755d
+spec:
+  apply: IfReady
```
Comment (ArnobKumarSaha) [inline docs/guides/redis/concepts/redisopsrequest.md:180]: `apply`, `timeout` fields should be described

File: docs/guides/redis/autoscaler/compute/redis.md [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+- `spec.compute.replicaset.resourceDiffPercentage` specifies the minimum resource difference in percentage. The default is 10%.
+  If the difference between current & recommended resource is less than ResourceDiffPercentage, Autoscaler Operator will ignore the updating.
+- `spec.compute.standalone.minAllowed` specifies the minimum allowed resources for the database.
+- `spec.compute.standalone.maxAllowed` specifies the maximum allowed resources for the database.
+- `spec.compute.standalone.controlledResources` specifies the resources that are controlled by the autoscaler.
+- `spec.compute.standalone.containerControlledValues` specifies which resource values should be controlled. The default is "RequestsAndLimits".
+- `spec.opsRequestOptions` contains the options to pass to the created OpsRequest. It has 3 fields. Know more about them here : [readinessCriteria](/docs/guides/redis/concepts/opsrequest.md#specreadinesscriteria), [timeout](/docs/guides/redis/concepts/opsrequest.md#spectimeout), [apply](/docs/guides/redis/concepts/opsrequest.md#specapply).
+
+If it was an `InMemory database`, we could also autoscaler the inMemory resources using Redis compute autoscaler, like below.
```
Comment (ArnobKumarSaha) [inline docs/guides/redis/autoscaler/compute/redis.md]: Remove lines 179-195. These are mongo specific

File: docs/guides/redis/concepts/catalog.md [hunk 3]
```diff
@@ -67,10 +84,19 @@ The default value of this field is `false`. If `spec.deprecated` is set to `true
...
 
+### spec.initContainer.image
+
+`spec.initContainer.image` is a required field that specifies the image for init container.
+
 ### spec.exporter.image
 
 `spec.exporter.image` is a required field that specifies the image which will be used to export Prometheus metrics.
 
+### spec.stash
```
Comment (ArnobKumarSaha) [inline docs/guides/redis/concepts/catalog.md:95]: There is anothe field. `spec.upgradeConstraints` describe that

File: docs/guides/redis/concepts/redis.md [hunk 4]
```diff
@@ -247,16 +384,28 @@ When `terminationPolicy` is `DoNotTerminate`, KubeDB takes advantage of `Validat
...
+### spec.halted
+Indicates that the database is halted and all offshoot Kubernetes resources except PVCs are deleted.
+
+## spec.healthChecker
+It defines the attributes for the health checker.
+- `spec.healthChecker.periodSeconds` specifies how often to perform the health check.
+- `spec.healthChecker.timeoutSeconds` specifies the number of seconds after which the probe times out.
+- `spec.healthChecker.failureThreshold` specifies minimum consecutive failures for the healthChecker to be considered failed.
+- `spec.healthChecker.disableWriteCheck` specifies whether to disable the writeCheck or not.
+
```
Comment (ArnobKumarSaha) [inline docs/guides/redis/concepts/redis.md:397]: Double check the fields of `spec.podTemplate` & `spec.serviceTemplate`. Some may have missed.

File: docs/guides/redis/scaling/horizontal-scaling/sentinel.md [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+$ kubectl get redissentinel -n demo sen-sample -o json | jq '.spec.replicas'
+3
+```
+
+The above output verifies that we have successfully scaled up the resources of the sentinel instance.
+### Horizontal Scale Redis
+
+Here, we are going to update the resources of the redis database to meet the desired resources after scaling.
+
+#### Create RedisOpsRequest:
```
Comment (ArnobKumarSaha) [inline docs/guides/redis/scaling/horizontal-scaling/sentinel.md:228]: HorizontalScaling of Redis has already been discussed in `cluster` page.   As It doesn't matter whether that Redis is referred by a Sentinel or not , We can say that(or refer to) in Cluster page. 

Remove repeatitive parts

File: docs/guides/redis/volume-expansion/volume-expansion.md [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+    mode: "Online"
+    redis: 2Gi
+```
+
+Here,
+
+- `spec.databaseRef.name` specifies that we are performing volume expansion operation on `sample-redis` database.
+- `spec.type` specifies that we are performing `VolumeExpansion` on our database.
+- `spec.volumeExpansion.redis` specifies the desired volume size.
+- `spec.volumeExpansion.mode` specifies the desired volume expansion mode (`Online` or `Offline`). Storageclass `topolvm-provisioner` supports `Online` volume expansion.
```
Comment (ArnobKumarSaha) [inline docs/guides/redis/volume-expansion/volume-expansion.md:151]: Differentiating between Online & Offline would be useful for users

**Review by ArnobKumarSaha — CHANGES_REQUESTED** [review]

File: docs/guides/redis/concepts/catalog.md [hunk 7]
```diff
@@ -67,10 +84,23 @@ The default value of this field is `false`. If `spec.deprecated` is set to `true
...
+
+This holds the Backup & Restore task definitions, where a `TaskRef` has a `Name` & `Params` section. Params specifies a list of parameters to pass to the task.
```
Comment (TasdidurRahman) [inline docs/guides/redis/concepts/catalog.md:97]: add stash link maybe!

File: docs/guides/redis/concepts/autoscaler.md [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+
+`RedisSentinelAutoscaler` does not have `spec.stoage` section. 
+
+All of them has the following sub-fields:
+
+- `trigger` indicates if storage autoscaling is enabled for this component of the database. If "On" then storage autoscaling is enabled. If "Off" then storage autoscaling is disabled.
+- `usageThreshold` indicates usage percentage threshold, if the current storage usage exceeds then storage autoscaling will be triggered.
+- `scalingThreshold` indicates the percentage of the current storage that will be scaled.
+- `expansionMode` indicates the volume expansion mode.
+
```
Comment (TasdidurRahman) [inline docs/guides/redis/concepts/autoscaler.md:142]: give example ?

File: docs/guides/redis/concepts/redisopsrequest.md [hunk 9]
```diff
@@ -0,0 +1,284 @@
+---
```
Comment (TasdidurRahman) [inline docs/guides/redis/concepts/redisopsrequest.md:1]: should go before autoscaler

**Review by TasdidurRahman — CHANGES_REQUESTED** [review]

File: docs/guides/redis/clustering/overview.md [hunk 10]
```diff
@@ -14,15 +14,28 @@ section_menu_id: guides
...
+- **Consistent Data**: Redis Cluster provides a distributed system for managing and maintaining the cluster, ensuring that data is always consistent and available.
+
+- **Easy Administration**: Redis Cluster provides a centralized management interface for the cluster, making it easier to manage and monitor large-scale Redis setups.
+
+- **Fast Performance**: Redis Cluster provides fast, in-memory data access, enabling fast and responsive applications.
 
+- **Simplified operations**: Redis Cluster eliminates the need for manual sharding, enabling a simpler and more automated way to scale Redis.
 ![redis-cluster](/docs/images/redis/redis-cluster.png)
 
 > Image reference [here](https://redislabs.com/redis-features/redis-cluster).
```
Comment (ArnobKumarSaha) [inline docs/guides/redis/clustering/overview.md]: this reference is redirecting to redis.com, not to an image.

File: docs/guides/redis/sentinel/overview.md [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+- **Load balancing**: Sentinel can be used to manage multiple Redis instances and provide load balancing, distributing clients across multiple Redis instances for better performance and reliability.
+
+- **Scalability**: Redis Sentinel can be used to create a scalable Redis setup, allowing multiple Redis instances to be added and managed as needed.
+
+- **Simplified administration**: Redis Sentinel provides a centralized management interface for Redis instances, making it easier to manage and monitor large-scale Redis setups.
+
+
+
+
+![redis-sentinel](/docs/images/redis/redis-sentinel.png)
```
Comment (ArnobKumarSaha) [inline docs/guides/redis/sentinel/overview.md:39]: Give the reference .

File: docs/guides/redis/sentinel/replacesentinel/_index.md [hunk 12]
```diff
@@ -0,0 +1,10 @@
+---
+title: Replace Sentinel
```
Comment (ArnobKumarSaha) [inline docs/guides/redis/sentinel/replacesentinel/_index.md:2]: Wont it be good if we move `Replace Sentinel` out of the sentinel folder, & treat it as same as other opsRequests ?

File: docs/guides/redis/autoscaler/compute/redis.md [hunk 13]
```diff
@@ -0,0 +1,363 @@
...
+- `spec.compute.replicaset.resourceDiffPercentage` specifies the minimum resource difference in percentage. The default is 10%.
+  If the difference between current & recommended resource is less than ResourceDiffPercentage, Autoscaler Operator will ignore the updating.
+- `spec.compute.standalone.minAllowed` specifies the minimum allowed resources for the database.
+- `spec.compute.standalone.maxAllowed` specifies the maximum allowed resources for the database.
+- `spec.compute.standalone.controlledResources` specifies the resources that are controlled by the autoscaler.
+- `spec.compute.standalone.containerControlledValues` specifies which resource values should be controlled. The default is "RequestsAndLimits".
+- `spec.opsRequestOptions` contains the options to pass to the created OpsRequest. It has 3 fields. Know more about them here : [readinessCriteria](/docs/guides/redis/concepts/opsrequest.md#specreadinesscriteria), [timeout](/docs/guides/redis/concepts/opsrequest.md#spectimeout), [apply](/docs/guides/redis/concepts/opsrequest.md#specapply).
```
Comment (ArnobKumarSaha) [inline docs/guides/redis/autoscaler/compute/redis.md]: links has not been correctly referred .

**Review by ArnobKumarSaha — CHANGES_REQUESTED** [review]

File: docs/guides/redis/autoscaler/compute/redis.md (hunk 13 above)
Comment (ArnobKumarSaha) [reply docs/guides/redis/autoscaler/compute/redis.md]: Resolve the build error.  That is all about `check links`

File: docs/guides/redis/sentinel/overview.md (hunk 11 above)
Comment (AbdullahAlShaad) [reply docs/guides/redis/sentinel/overview.md:39]: Collected from google

File: docs/guides/redis/clustering/overview.md (hunk 10 above)
Comment (AbdullahAlShaad) [reply docs/guides/redis/clustering/overview.md]: Removed Ref

File: docs/guides/redis/sentinel/replacesentinel/_index.md (hunk 12 above)
Comment (AbdullahAlShaad) [reply docs/guides/redis/sentinel/replacesentinel/_index.md:2]: This Ops Request is for a specific Mode. So kept in under that Mode.

**Review by ArnobKumarSaha — DISMISSED** [review]

**Review by heheh13 — DISMISSED** [review]

**Pull request merged**